### PR TITLE
AQA DAMAGE RUN LOGIC

### PR DESCRIPTION
**Pull request by SchwartzGandhi**
500 million years in the making

### CODE REVIEW

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 1]
```diff
@@ -1208,8 +1286,23 @@ Extra - room_id: [28]
...
+      All of the following:
+          Morph Ball
+          Any of the following:
+              After Sector 4 Water Level Lowered
+              All of the following:
+                  # Damage Reqs
+                  Damage Runs (Beginner)
+                  Any of the following:
+                      Electrified Water Damage ≥ 16
+                      Gravity Suit and Electrified Water Damage ≥ 3
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1305]: are we able to combine the reqs to go *to* the item, for generators sake?

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 2]
```diff
@@ -1351,27 +1453,36 @@ Extra - room_id: [33]
...
 > Behind Speed Blocks; Heals? False
   * Layers: default
   > Event - Pump Control Terminal
       Trivial
   > Next to Pickup
-      Morph Ball
+      All of the following:
+          Morph Ball
+          Any of the following:
+              After Sector 4 Water Level Lowered
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1479]: After Sector 4 Water Level Lowered or Generator Workarounds

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 3]
```diff
@@ -1395,15 +1506,25 @@ that you were able to collect the event.
...
+              All of the following:
+                  # Damage Reqs
+                  Damage Runs (Beginner)
+                  Any of the following:
+                      Electrified Water Damage ≥ 44
+                      Gravity Suit and Electrified Water Damage ≥ 15
   > Behind Speed Blocks
       Any of the following:
-          Can Use Springball
+          After Sector 4 Water Level Lowered or Can Use Springball
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1518]: does spring ball with the water really not get high enough?

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 4]
```diff
@@ -1562,14 +1683,15 @@ Extra - room_id: [36]
               # Navigate through water
               After Sector 4 Water Level Lowered
               Any of the following:
-                  # TODO: damage numbers
-                  Damage Runs (Intermediate)
-                  Gravity Suit and Damage Runs (Beginner)
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1567]: is this to get to the item and go back?

Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:1567]: correct

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 5]
```diff
@@ -1514,17 +1495,21 @@ that you were able to collect the event.
...
-                  Gravity Suit and Electrified Water Damage ≥ 9
+      All of the following:
+          Any of the following:
+              # Get into tunnel
+              Can Use Springball
+              All of the following:
+                  Can Use Bombs
+                  Gravity Suit or After Sector 4 Water Level Lowered or Enabled Generator Workaround
+          Any of the following:
+              After Sector 4 Water Level Lowered
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1506]: this also needs the generator misc resource.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 6]
```diff
@@ -97,12 +97,25 @@ Extra - room_id: [3]
               Diffusion Missile Data and Missiles
               All of the following:
                   # Freeze Middle Powamp
-                  Ice Missile Data and Missiles and Movement (Advanced)
-                  After Sector 4 Water Level Lowered or Damage Runs (Beginner)
+                  Knowledge (Advanced) and Movement (Advanced)
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:100]: this is now knowledge advancde + movement advanced, while the other is only knowledge advanced. they should be synced.

**Review by Miepee — APPROVED** [review]

**Pull request merged**